coderd/taskname/taskname.go

```diff
@@ -0,0 +1,99 @@
+package taskname
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/anthropics/anthropic-sdk-go"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/aisdk-go"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+const systemPrompt = `Generate a short workspace name from this AI task prompt.
+
+Requirements:
+- Only lowercase letters, numbers, and hyphens
+- Start with "task-"
+- End with a random number between 0-99
+- Maximum 32 characters total
+- Descriptive of the main task
+
+Examples:
+- "Help me debug a Python script" → "task-python-debug-12"
+- "Create a React dashboard component" → "task-react-dashboard-93"
+- "Analyze sales data from Q3" → "task-analyze-q3-sales-37"
+- "Set up CI/CD pipeline" → "task-setup-cicd-44"
+
+If you cannot create a suitable name:
+- Respond with "task-workspace"
+- Do not end with a random number`
+
+func Generate(ctx context.Context, prompt, fallback string) (string, error) {
+	conversation := []aisdk.Message{
+		{
+			Role: "system",
+			Parts: []aisdk.Part{{
+				Type: aisdk.PartTypeText,
+				Text: systemPrompt,
+			}},
+		},
+		{
+			Role: "user",
+			Parts: []aisdk.Part{{
+				Type: aisdk.PartTypeText,
+				Text: prompt,
+			}},
+		},
+	}
+
+	if apiKey := os.Getenv("ANTHROPIC_API_KEY"); apiKey == "" {
+		return fallback, nil
+	}
+
+	anthropicClient := anthropic.NewClient(anthropic.DefaultClientOptions()...)
+
+	stream, err := anthropicDataStream(ctx, anthropicClient, conversation)
+	if err != nil {
+		return fallback, xerrors.Errorf("create anthropic data stream: %w", err)
```

Comment:
Nit: Returning a value and error is not common practice and may be surprising to the reader. The way fallback is used here it would make sense to either return error or provide the fallback, but not both. Perhaps returning empty strings in all fallback cases and handling it at the call-site would be preferable?